sycl/test-e2e/KernelCompiler/kernel_compiler_spirv.cpp

```diff
@@ -0,0 +1,158 @@
+  spv_stream.read(reinterpret_cast<char *>(spv.data()), sz);
+
+  // Create a kernel bundle from the binary SPIR-V.
+  sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source> kb_src =
+      syclex::create_kernel_bundle_from_source(
+          q.get_context(), syclex::source_language::spirv, spv);
+
+  // Build the SPIR-V module for our device.
+  sycl::kernel_bundle<sycl::bundle_state::executable> kb_exe =
+      syclex::build(kb_src);
+  return kb_exe;
+}
+
+void testSimpleKernel(sycl::queue &q, const sycl::kernel &kernel,
+                      int multiplier, int added) {
+  const auto num_args = kernel.get_info<sycl::info::kernel::num_args>();
+  assert(num_args == 2 && "kernel should take 2 args");
+
+  constexpr int N = 4;
+  std::array<int, N> input_array{0, 1, 2, 3};
+  std::array<int, N> output_array{};
+
+  sycl::buffer input_buffer(input_array.data(), sycl::range<1>(N));
+  sycl::buffer output_buffer(output_array.data(), sycl::range<1>(N));
+
+  q.submit([&](sycl::handler &cgh) {
+    cgh.set_arg(0, input_buffer.get_access<sycl::access::mode::read>(cgh));
+    cgh.set_arg(1, output_buffer.get_access<sycl::access::mode::write>(cgh));
+    cgh.parallel_for(sycl::range<1>{N}, kernel);
+  });
+
+  sycl::host_accessor out{output_buffer};
+  for (int i = 0; i < N; i++) {
+    assert(out[i] == ((i * multiplier) + added));
+  }
+}
+
+template <typename T>
+void testParam(sycl::queue &q, const sycl::kernel &kernel) {
+  const auto num_args = kernel.get_info<sycl::info::kernel::num_args>();
+  assert(num_args == 4 && "kernel should take 4 args");
+
+  // Kernel computes sum of squared inputs.
+  const T a = 2;
+  const T b = 5;
+  const T expected = (a * a) + (b * b);
+
+  sycl::buffer a_buffer(&a, sycl::range<1>(1));
+
+  T *const b_ptr = sycl::malloc_shared<T>(1, q);
+  b_ptr[0] = b;
+
+  T output{};
+  sycl::buffer output_buffer(&output, sycl::range<1>(1));
+
+  q.submit([&](sycl::handler &cgh) {
+    sycl::local_accessor<T, 1> local(1, cgh);
+    // Pass T for scalar parameter.
+    cgh.set_arg(0, a);
+    // Pass USM pointer for OpTypePointer(CrossWorkgroup) parameter.
+    cgh.set_arg(1, b_ptr);
+    // Pass sycl::accessor for OpTypePointer(CrossWorkgroup) parameter.
+    cgh.set_arg(
+        2, output_buffer.template get_access<sycl::access::mode::write>(cgh));
+    // Pass sycl::local_accessor for OpTypePointer(Workgroup) parameter.
+    cgh.set_arg(3, local);
+    cgh.parallel_for(sycl::range<1>{1}, kernel);
+  });
+
+  sycl::host_accessor out{output_buffer};
+  assert(out[0] == expected);
+  sycl::free(b_ptr, q);
+}
+
+void testKernelsFromSpvFile(std::string kernels_file,
+                            std::string fp16_kernel_file,
+                            std::string fp64_kernel_file) {
+  const auto getKernel =
+      [](sycl::kernel_bundle<sycl::bundle_state::executable> &bundle,
+         const std::string &name) {
+        return bundle.ext_oneapi_get_kernel(name);
+      };
+
+  sycl::queue q;
```

Comment:
if you are having test failures on PVC, it is because there is a bug where this gets set up with multiple devices, not just one.  L0 will then refuse to instantiate the zeModule/zeProgram  ("one device max" or some such)
The workaround is to do this:
```
  sycl::device d;
  sycl::context ctx{d};
  sycl::queue q{ctx, d};
```

until that gets fixed.